lib/capistrano/tasks/maintenance.rake

```diff
@@ -76,13 +76,19 @@ namespace :maintenance do
       info ""
       info "Add the \e[35mfollowing configuration\e[0m to your Apache \e[35m.htaccess\e[0m file:"
       info "=============================="
+      info "<LocationMatch \"\\.(css|gif|ico|jpe?g|js|png|svg|webp|webm)$\">"
+      info "    ErrorDocument 503 \"Service unavailable.\""
+      info "</LocationMatch>"
       info "<IfModule mod_rewrite.c>"
-      info "RewriteEngine On"
-      info "RewriteCond %{DOCUMENT_ROOT}/#{fetch(:maintenance_basename)} -f"
-      info "RewriteCond %{SCRIPT_FILENAME} !#{fetch(:maintenance_basename)}"
-      info "RewriteRule ^.*$ /#{fetch(:maintenance_basename)} [R=503,L]"
-      info "ErrorDocument 503 /#{fetch(:maintenance_basename)}"
-      info "Header Set Cache-Control \"max-age=0, no-store\""
+      info "    <If \"! %{REQUEST_URI} =~ /\\.(css|gif|ico|jpe?g|js|png|svg|webp|webm)$/i && -f '%{DOCUMENT_ROOT}/#{fetch(:maintenance_basename)}'\">"
+      info "        ErrorDocument 503 /#{fetch(:maintenance_basename)}"
+      info "    </If>"
+      info "    RewriteEngine On"
+      info "    RewriteCond \"%{ENV:REDIRECT_STATUS}\" !=503"
+      info "    RewriteCond %{DOCUMENT_ROOT}/#{fetch(:maintenance_basename)} -f"
+      info "    RewriteCond %{SCRIPT_FILENAME} !#{fetch(:maintenance_basename)}"
+      info "    RewriteRule \".*\" \"-\" [R=503,L,E=NOCACHE:1]"
+      info "    Header always set Cache-Control \"no-store, no-cache, must-revalidate\" env=REDIRECT_NOCACHE"
```

Comment:
Can we use a more namespace env variable like `REDIRECT_MAINTENANCE_NOCACHE` ?

Reply:
👍

Reply:
I don't see why we will have an env in a redirection named NOCACHE that has another effect than disable cache.  But if you want we can rename it, I don't mind.

Reply:
OK for using `NOCACHE`